packages/sanity-plugin-nacelle-input/src/components/NacelleLinker.js

```diff
@@ -0,0 +1,307 @@
+import React, { useState, useEffect, useCallback, useContext } from 'react'
+import PropTypes from 'prop-types'
+
+import PatchEvent, { set, unset } from 'part:@sanity/form-builder/patch-event'
+import FormField from 'part:@sanity/components/formfields/default'
+import { useId } from '@reach/auto-id'
+
+import config from 'config:@nacelle/sanity-plugin-nacelle-input'
+
+import {
+  ThemeProvider,
+  studioTheme,
+  Box,
+  TextInput,
+  Button,
+  Dialog,
+  Select,
+  Tab,
+  TabList,
+  TabPanel,
+  Stack,
+  Flex
+} from '@sanity/ui'
+import NacelleDataFetcher from './NacelleDataFetcher'
+import { GET_PRODUCTS, GET_COLLECTIONS } from '../queries'
+import {
+  HandleContext,
+  SearchQueryContext,
+  SpaceOptionsContext
+} from '../context'
+
+const createPatchFrom = (value) =>
+  PatchEvent.from(value === '' ? unset() : set(value))
+
+const NacelleData = ({ dataType, active, searchTerm }) => {
+  const { spaceOptions } = useContext(SpaceOptionsContext)
+  switch (dataType) {
+    case 'products':
+      return (
+        <NacelleDataFetcher
+          query={GET_PRODUCTS}
+          options={spaceOptions}
+          className="tabContent"
+          active={active}
+          id="products-panel"
+          type="products"
+          searchTerm={searchTerm}
+        />
+      )
+    case 'collections':
+      return (
+        <NacelleDataFetcher
+          query={GET_COLLECTIONS}
+          options={spaceOptions}
+          className="tabContent"
+          active={active}
+          id="collections-panel"
+          type="productCollections"
+          searchTerm={searchTerm}
+        />
+      )
+  }
+}
+
+NacelleData.propTypes = {
+  dataType: PropTypes.string.isRequired,
+  active: PropTypes.bool,
+  searchTerm: PropTypes.string
+}
+
+const SearchIcon = () => (
+  <svg
+    data-sanity-icon="search"
+    width="1em"
+    height="1em"
+    viewBox="0 0 25 25"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <circle
+      cx="10.5"
+      cy="10.5"
+      r="5"
+      stroke="currentColor"
+      strokeWidth="1.2"
+    ></circle>
+    <path d="M14 14L20 20" stroke="currentColor" strokeWidth="1.2"></path>
+  </svg>
+)
+
+const Interface = ({
+  dataType,
+  interfaceOpen,
+  children,
+  activeTab,
+  setActiveTab
+}) => {
+  const { spaceOptions, setSpaceOptions } = useContext(SpaceOptionsContext)
+
+  const dataTypes = Array.isArray(dataType) ? dataType.sort() : [dataType]
+  const multiTab = dataTypes.length > 1
+
+  const multiSelect =
+    Array.isArray(config.nacelleSpaces) &&
+    config.nacelleSpaces.length > 1 &&
+    config.nacelleSpaces.some(
+      (s) => s.nacelleEndpoint && s.nacelleSpaceToken && s.spaceName
+    )
+
+  const onSelect = (e) => {
+    const activeSpace = config.nacelleSpaces.find(
+      (space) => space.nacelleEndpoint == e.target.value
+    )
+    setSpaceOptions(activeSpace)
+  }
```

Comment:
future work:
We might want to configure clearing the data when `setSpaceOptions` is called. That way stale products don't stay while the fresh data gets loaded